lib/scope.js

```diff
@@ -119,38 +119,26 @@ Scope.prototype.delete = function _delete(uri, requestBody, options) {
 };
 
 Scope.prototype.pendingMocks = function pendingMocks() {
-  return Object.keys(this.keyedInterceptors);
+  var self = this;
+
+  var pendingInterceptorKeys = Object.keys(this.keyedInterceptors).filter(function (key) {
+    var interceptorList = self.keyedInterceptors[key];
+    var pendingInterceptors = interceptorList.filter(function (interceptor) {
+      var persistedAndUsed = self._persist && interceptor.interceptionCounter > 0;
+      return !persistedAndUsed && !interceptor.optional;
+    });
+    return pendingInterceptors.length > 0;
+  });
+
+  return pendingInterceptorKeys;
```

Comment:
I see that `pendingMocks` was incorrectly returning all mocks. I expect that some users of nock may have depended of that behaviour.

Would you mind adding something like `allMocks`/`remainingMocks`/`activeMocks` (naming's hard) that's doing what `pendingMocks` was previously doing?